File: api/web/models/chat_session.py
"""APIs for ChatSession"""
from typing import Any, Dict, Optional
from modules.types import PMObject


class ChatSessionInstance(PMObject):
    id: int
    created_at: str
    uuid: str
    run_from_deployment: bool

    session_metadata: Optional[Dict[str, Any]] = None

    version_uuid: str

File: frontend/src/types/ChatSession.ts
// Interfaces for ChatSession retrieval processes

/**
 * Interface for request to read ChatModelVersion's ChatSessions.
 */
export interface ReadChatSessionsRequest {
  chat_model_version_uuid: string;
}

/**
 * General interface for ChatSession.
 */
export interface ChatSession {
  id: number;
  created_at: string;
  uuid: string;
  run_from_deployment: boolean;
  session_metadata: Record<string, any>;
  version_uuid: string;
}
